monkey/monkey_island/cc/resources/agent_signals.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from http import HTTPStatus
+from json import JSONDecodeError
+
+from flask import request
+
+from monkey_island.cc.event_queue import IIslandEventQueue, IslandEventTopic
+from monkey_island.cc.resources.AbstractResource import AbstractResource
+from monkey_island.cc.services import AgentSignalsService
+
+logger = logging.getLogger(__name__)
+
+
+class AgentSignals(AbstractResource):
+    urls = ["/api/agent-signals/terminate-all", "/api/agent-signals/<string:agent_id>"]
```

Comment:
We should break this into two resources. One for `POST agent-signals/terminate-all` and another for `GET agent-signals/<string:agent_id>`. `GET agent-signals/terminate-all` and `POST agent-signals/<string:agent_id>` are not valid operations, but with this implementation the resource will attempt to service such requests.

Reply:
+1

Reply:
I suggest we have an RPC-style endpoint for terminating all agents (`/api/terminate-all-agents`). The POST function doesn't have anything to do with `AgentSignals` anyway.